qiskit_experiments/library/characterization/analysis/stark_ramsey_fast_analysis.py

```diff
@@ -0,0 +1,303 @@
+                "Yneg": {"series": "Y", "direction": "neg"},
+            },
+            plotter=ramsey_plotter,
+        )
+
+        return options
+
+    def _generate_fit_guesses(
+        self,
+        user_opt: curve.FitOptions,
+        curve_data: curve.CurveData,
+    ) -> Union[curve.FitOptions, List[curve.FitOptions]]:
+        """Create algorithmic initial fit guess from analysis options and curve data.
+
+        Args:
+            user_opt: Fit options filled with user provided guess and bounds.
+            curve_data: Formatted data collection to fit.
+
+        Returns:
+            List of fit options that are passed to the fitter function.
+        """
+        # Compute offset guess
+        user_opt.p0.set_if_empty(
+            offset=np.mean(curve_data.y),
+            f_err=0.0,
+        )
+        user_opt.bounds.set_if_empty(
+            offset=(-1, 1),
+            amp=(0, 1),
+        )
+        est_offs = user_opt.p0["offset"]
+
+        # Compute amplitude guess
+        amps = np.zeros(0)
+        for direction in ("pos", "neg"):
+            ram_x_off = curve_data.get_subset_of(f"X{direction}").y - est_offs
+            ram_y_off = curve_data.get_subset_of(f"Y{direction}").y - est_offs
+            amps = np.concatenate([amps, np.sqrt(ram_x_off**2 + ram_y_off**2)])
+        user_opt.p0.set_if_empty(amp=np.median(amps))
+        est_a = user_opt.p0["amp"]
+        d_const = user_opt.p0["dt"]
+
+        # Compute polynominal coefficients
+        guesses = []
+        for direction in ("pos", "neg"):
+            ram_x_data = curve_data.get_subset_of(f"X{direction}")
+            ram_y_data = curve_data.get_subset_of(f"Y{direction}")
+            xvals = ram_x_data.x
+
+            # Get normalized sinusoidals
+            xnorm = (ram_x_data.y - est_offs) / est_a
+            ynorm = (ram_y_data.y - est_offs) / est_a
+
+            # Compute derivative to extract polynominals from sinusoidal
+            dx = np.diff(xnorm) / np.diff(xvals)
+            dy = np.diff(ynorm) / np.diff(xvals)
+
+            # Eliminate sinusoidal
+            phase_poly = np.sqrt(dx**2 + dy**2)
+
+            tmp = []
+            # Consider both signs to expand square root.
+            for sign in (1, -1):
+                # Do polyfit up to 2rd order.
+                # This must correspond to the 3rd order in the original function.
+                vmat_xpoly = np.vstack((xvals[1:] ** 2, xvals[1:], np.ones(xvals.size - 1))).T
+                coeffs = np.linalg.lstsq(vmat_xpoly, sign * phase_poly, rcond=-1)[0]
+
+                poly_guess = {
+                    f"c1_{direction}": coeffs[2] / 1 / d_const,
+                    f"c2_{direction}": coeffs[1] / 2 / d_const,
+                    f"c3_{direction}": coeffs[0] / 3 / d_const,
+                }
+                tmp.append(poly_guess)
+            guesses.append(tmp)
+
+        fit_opts = []
+        for guess_pos, guess_neg in product(*guesses):
```

Comment:
Minor: Could you look at the residuals to choose the better sign so you have only one guess instead of four?

Reply:
This is simplified in [8dee4eb](https://github.com/Qiskit-Extensions/qiskit-experiments/pull/1009/commits/8dee4ebf3a645937466770b0ea50857af81aa2be). Note that negative sign corresponds to unphysical setting that induces positive (negative) Stark shift with negative (positive) Stark amplitude. I did bit of overengineering for unittest with unphysical parameters, which is also removed. Parameter bound is also added to the quadratic terms.